packages/react-storage/src/components/StorageBrowser/context/actions/__tests__/listLocationsAction.spec.ts

```diff
@@ -82,11 +82,15 @@ describe('createListLocationsAction', () => {
     // assume, total items: 1500; default page limit: 1000
     mockListLocations.mockResolvedValueOnce({
       locations: generateMockLocations(600),
-      nextToken: 'next',
+      nextToken: 'next-1',
+    });
+    mockListLocations.mockResolvedValueOnce({
+      locations: generateMockLocations(200),
+      nextToken: 'next-2',
     });
     mockListLocations.mockResolvedValueOnce({
-      locations: generateMockLocations(400),
-      nextToken: 'next-oooo',
+      locations: generateMockLocations(200),
```

Comment:
This issue only occurs when it's recursively called more than 2 times.
However we can further improve the test, will raise another PR